README.md

```diff
@@ -65,6 +65,81 @@ Concretely the following items are on the road map:
 
 This crate's minimum supported rustc version is 1.66.0.
 
+## Working on this project
+
+Due to the large number of native dependencies, this project uses [Nix](https://nixos.org/) and [direnv](https://direnv.net/) to streamline the development experience.
+
+### Setting up your environment
+
+For the best experience, please follow these instructions to setup your environment:
+1. Install Nix following [their guide](https://nixos.org/download.html) for your operating system
+2. Create the file `~/.config/nix/nix.conf` with the contents:
+```ini
+experimental-features = nix-command
+extra-experimental-features = flakes
+```
+3. Install direnv into your Nix profile by running:
+```sh
+nix profile install nixpkgs#direnv
+```
+4. Add direnv to your shell following [their guide](https://direnv.net/docs/hook.html)
+5. Restart your shell
+
+### Shell & editor experience
+
+Now that your environment is set up, you can get to work on the project.
+
+1. Clone the repository, such as:
+```sh
+git clone git@github.com:noir-lang/noir
+```
+2. Navigate to the directory:
+```sh
+cd noir
+```
+3. You should see a __direnv error__ because projects aren't allowed by default. Make sure you've reviewed and trust our `.envrc` file, then you need to run:
```

Comment:
Not a blocker for this PR: should we have a central place where we put nix and direnv instructions? It seems like we will have duplicated information between all of our repositories which will use nix.

Reply:
#1199